config/config-api/src/com/thoughtworks/go/util/XmlUtils.java

```diff
@@ -95,4 +98,16 @@ public static boolean doesNotMatchUsingXsdRegex(Pattern pattern, String textToMa
     public static boolean matchUsingRegex(Pattern pattern, String textToMatch) {
         return pattern.matcher(textToMatch).matches();
     }
+
+    public static String stripProlog(String line) {
+        if (line.contains(PROLOG_START)) {
```

Comment:
@sghill - Like I'd said in [my comment](https://github.com/gocd/gocd/pull/1570#discussion_r42316842) before -

Ideally what you'd like to look for is line that `startsWith` ignoring any leading whitespaces that lead upto the prolog.

This is needed because of files that possibly look like this (formatted for readability, but could be all on one line) would be parsed incorrectly with the implementation that you have. Perhaps it's just a matter of calling `strip` on the line before parsing it?

```xml
<testsuite>
  <testcase>
    <stdout>
      <![CDATA[....<?xml version="1.0" encoding="utf-8"?>...]]>
    </stdout>
  </testcase>
</testsuite>
```